modules/exploits/windows/local/lexmark_driver_privesc.rb

```diff
@@ -0,0 +1,138 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = NormalRanking
+
+  include Msf::Post::File
+  include Msf::Exploit::EXE
+  include Msf::Post::Windows::Priv
+  include Msf::Exploit::FileDropper
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Lexmark Driver Privilege Escalation',
+        'Description' => %q{
+          Various Lexmark printer drivers allow escalation of
+          privileges on Windows systems.
+
+          For vulnerable drivers, a low-privileged user can
+          modify an XML configuration file and replace a
+          DLL path with a malicious DLL path.
+
+          `PrintIsolationHost.exe`, a Windows process running
+          as NT AUTHORITY\SYSTEM, loads driver-specific DLLs
+          during the installation of a printer. A user can
+          elevate to SYSTEM by writing a malicious DLL to
+          the vulnerable configuration file and adding a new
+          printer with a vulnerable driver.
+
+          This module leverages the `prnmngr.vbs` script
+          to add and delete printers.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Jacob Baines', # discovery, PoC, module
+          'Shelby Pace' # original Ricoh driver module
+        ],
+        'References' =>
+          [
+            [ 'CVE', '2021-35449'],
+            [ 'URL', 'http://support.lexmark.com/index?page=content&id=TE953']
+          ],
+        'Arch' => [ ARCH_X86, ARCH_X64 ],
+        'Platform' => 'win',
+        'Payload' =>
+        {
+        },
+        'SessionTypes' => [ 'meterpreter' ],
+        'Targets' =>
+          [
+            [
+              'Windows', { 'Arch' => [ ARCH_X86, ARCH_X64 ] }
+            ]
+          ],
+        'Notes' =>
+        {
+          'SideEffects' => [ ARTIFACTS_ON_DISK ],
+          'Reliability' => [ REPEATABLE_SESSION ],
+          'Stability' => [ SERVICE_RESOURCE_LOSS ]
+        },
+        'DisclosureDate' => '2021-07-15',
+        'DefaultTarget' => 0
+      )
+    )
+
+    self.needs_cleanup = true
```

Comment:
This doesn't look like it's being used anywhere. Can it be removed?

Reply:
Nevermind, this is used by some of the library code to identify that this modules needs to have it's cleanup method run. Marking this as resolved.